10up/ElasticPress
# Facet by Meta Range block UI.

### Description of the Change

Adds a range slider UI for the Facet by Meta Range block.

![Screenshot 2023-02-23 at 5 57 40 pm](https://user-images.githubusercontent.com/4100733/220839017-83f4de65-dc74-4dcb-b2fc-9096818bebb1.png)

Also updates the block preview with a preview of the slider, a message when a preview is unavailable, and a wizard for selecting a field when adding the block for the first time. The wizard is required to avoid a preview unavailable message being the first thing the user sees, and allows that message to include the field name since the wizard can be used when no field is selected.

https://user-images.githubusercontent.com/4100733/221603861-52c2a991-83f7-4815-a857-28cce3c0307d.mp4


Closes #3291

E2E tests [to come](https://github.com/10up/ElasticPress/issues/3357).

### How to test the Change
- Add the Facet by Range meta block.
  - When adding the block you should be prompted to choose a field.
  - After choosing a field you should see either:
    - A preview of the range slider. It should not be interactive.
    - A warning that the selected field is not numeric and cannot be used for the range slider.
  - On the front end the range slider should be interactive, and when the Filter button is pressed it should filter results by the selected range.
  - It should be possible to undo the range filter by reselecting the widest range of values.  

### Changelog Entry
See #3289

### Credits
@JakePT @felipeelia 

### Checklist:
